LLVM.Imports.BitReader.pas

```diff
@@ -9,16 +9,25 @@ interface
   LLVM.Imports.Types;
 
 //LLVMParseBitcode is deprecated, therefore we implement only LLVMParseBitcode2, but as LLVMParseBitcode
-function LLVMParseBitcode(MemBuf: TLLVMMemoryBufferRef; out OutModule: TLLVMModuleRef): TLLVMBool; cdecl; external CLLVMLibrary name 'LLVMParseBitcode2';
+function LLVMParseBitcode(MemBuf: TLLVMMemoryBufferRef; out OutModule: TLLVMModuleRef; var OutMessage: PLLVMChar): TLLVMBool; cdecl; external CLLVMLibrary name 'LLVMParseBitcode2';
+//* Builds a module from the bitcode in the specified memory buffer, returning a  reference to the module via the OutModule parameter. Returns 0 on success. */
+function LLVMParseBitcode2(MemBuf: TLLVMMemoryBufferRef; out OutModule: TLLVMModuleRef): TLLVMBool; cdecl; external CLLVMLibrary name 'LLVMParseBitcode2';
 
 //LLVMParseBitcodeInContext is deprecated, therefore we implement only LLVMParseBitcodeInContext2, but as LLVMParseBitcodeInContext
-function LLVMParseBitcodeInContext(ContextRef: TLLVMContextRef; MemBuf: TLLVMMemoryBufferRef; out OutModule: TLLVMModuleRef): TLLVMBool; cdecl; external CLLVMLibrary name 'LLVMParseBitcodeInContext2';
+function LLVMParseBitcodeInContext(ContextRef: TLLVMContextRef; MemBuf: TLLVMMemoryBufferRef; out OutModule: TLLVMModuleRef;OutMessage: PLLVMChar): TLLVMBool; cdecl; external CLLVMLibrary name 'LLVMParseBitcodeInContext2';
+// but as LLVMParseBitcodeInContext
+function LLVMParseBitcodeInContext2(ContextRef: TLLVMContextRef; MemBuf: TLLVMMemoryBufferRef; out OutModule: TLLVMModuleRef): TLLVMBool; cdecl; external CLLVMLibrary name 'LLVMParseBitcodeInContext2';
 
 //same as with the other functions above, deprecated stuff etc
-function LLVMGetBitcodeModuleInContext(ContextRef: TLLVMContextRef; MemBuf: TLLVMMemoryBufferRef; out OutM: TLLVMModuleRef): TLLVMBool; cdecl; external CLLVMLibrary name 'LLVMGetBitcodeModuleInContext2';
+function LLVMGetBitcodeModuleInContext(ContextRef: TLLVMContextRef; MemBuf: TLLVMMemoryBufferRef; out OutM: TLLVMModuleRef;var OutMessage: PLLVMChar): TLLVMBool; cdecl; external CLLVMLibrary name 'LLVMGetBitcodeModuleInContext2';
+//** Reads a module from the specified path, returning via the OutMP parameter a module provider which performs lazy deserialization. Returns 0 on success. */
+function LLVMGetBitcodeModuleInContext2(ContextRef: TLLVMContextRef; MemBuf: TLLVMMemoryBufferRef; out OutM: TLLVMModuleRef): TLLVMBool; cdecl; external CLLVMLibrary name 'LLVMGetBitcodeModuleInContext2';
 
 //yada yada deprecated
-function LLVMGetBitcodeModule(MemBuf: TLLVMMemoryBufferRef; out OutM: TLLVMModuleRef): TLLVMBool; cdecl; external CLLVMLibrary name 'LLVMGetBitcodeModule2';
+function LLVMGetBitcodeModule(MemBuf: TLLVMMemoryBufferRef; out OutM: TLLVMModuleRef; var OutMessage: PLLVMChar): TLLVMBool; cdecl; external CLLVMLibrary name 'LLVMGetBitcodeModule2';
+
+function LLVMGetBitcodeModule2(MemBuf: TLLVMMemoryBufferRef; out OutM: TLLVMModuleRef): TLLVMBool; cdecl; external CLLVMLibrary name 'LLVMGetBitcodeModule2';
```

Comment:
S.O.

Reply:
yes exactly but it remains obsolete so I had eliminated it but then I reinserted it for compatibility.

Reply:
No this does not work.
LLVMGetBitcodeModule/LLVMGetBitcodeModule 2 are declared with 2 different signatures but both reference LLVMGetBitcodeModule2 from the dll. One of them is going to crash, because of faulty signature

Reply:
`LLVMBool LLVMGetBitcodeModule(LLVMMemoryBufferRef MemBuf, LLVMModuleRef *OutM,
                              char **OutMessage) {
  return LLVMGetBitcodeModuleInContext(LLVMGetGlobalContext(), MemBuf, OutM,
                                       OutMessage);
}

LLVMBool LLVMGetBitcodeModule2(LLVMMemoryBufferRef MemBuf,
                               LLVMModuleRef *OutM) {
  return LLVMGetBitcodeModuleInContext2(LLVMGetGlobalContext(), MemBuf, OutM);
}`

for me this work 
`if LLVMCreateMemoryBufferWithContentsOfFile('bitcode.bc',bufBC, msg) then
      begin
          var b,a : TLLVMBool;
          m.Value := nil;
          a := LLVMGetBitcodeModule(bufBC,m, msg);
          if a then
             b := LLVMGetBitcodeModule2(bufBC,m);
           if b then
              MessageBoxExA(0,'test pk','info',MB_OK,0) ;
      end;`

Reply:
This can NOT work. Both of the functions import the same using "name 'LLVMGetBitcodeModule2'" but have different signatures. If it is not crashing, it may damages other parts of the memory.

Reply:
No problem you can discard the change ;) but the functions are these with these parameters:

`/* This is deprecated. Use LLVMGetBitcodeModule2. */
LLVMBool LLVMGetBitcodeModule(LLVMMemoryBufferRef MemBuf, LLVMModuleRef *OutM,char **OutMessage);

LLVMBool LLVMGetBitcodeModule2(LLVMMemoryBufferRef MemBuf, LLVMModuleRef *OutM);`

`LLVMBool LLVMGetBitcodeModule(LLVMMemoryBufferRef MemBuf, LLVMModuleRef *OutM, char **OutMessage) 
{
  return LLVMGetBitcodeModuleInContext(LLVMGetGlobalContext(), MemBuf, OutM,  OutMessage);
}

LLVMBool LLVMGetBitcodeModule2(LLVMMemoryBufferRef MemBuf,LLVMModuleRef *OutM) 
{
  return LLVMGetBitcodeModuleInContext2(LLVMGetGlobalContext(), MemBuf, OutM);
}`

Reply:
I know but your imports are WRONG!

Reply:
I'm sorry but the bad English did not allow me to understand what you were referring to, I understood the function and not the import declaration. now I understand 😁😁

Reply:
Fix